File: 백준/Silver/10828. 스택/스택.py
import sys
from collections import deque


N = int(sys.stdin.readline().strip())

stack = deque()
for _ in range(N):
  word = sys.stdin.readline().strip().split(' ')
#   print(word)

  if word[0] == 'push':
    stack.append(int(word[1]))

  if word[0] == 'pop':
    if stack:
      print(stack.pop())
    else:
      print(-1)

  if word[0] == 'size':
    print(len(stack))

  if word[0] == 'empty':
    if stack:
      print(0)
    else:
      print(1)

  if word[0] == 'top':
    if stack:
      print(stack[-1])
    else:
      print(-1)
      

File: 백준/Silver/10828. 스택/스택.js
// /dev/stdin
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map((x) => x.replace("\r", ""));

const N = Number(input[0]);
const STACK = [];
let answer = "";

for (let i = 1; i <= N; i++) {
  const cmd = input[i];

  if (cmd.startsWith("push")) {
    const val = Number(cmd.split(" ")[1]);
    STACK.push(val);
    continue;
  }

  if (cmd === "pop") {
    answer += (STACK.length ? STACK.pop() : -1) + "\n";
    continue;
  }

  if (cmd === "size") {
    answer += STACK.length + "\n";
    continue;
  }

  if (cmd === "empty") {
    answer += (STACK.length === 0 ? 1 : 0) + "\n";
    continue;
  }

  if (cmd === "top") {
    answer += (STACK.length > 0 ? STACK[STACK.length - 1] : -1) + "\n";
    continue;
  }
}

console.log(answer);
